process/block/preprocess/transactionsV2.go

```diff
@@ -428,29 +470,183 @@ func (txs *transactions) createScheduledMiniBlocks(
 			continue
 		}
 
+		if len(senderAddressToSkip) > 0 {
+			if bytes.Equal(senderAddressToSkip, tx.GetSndAddr()) {
+				schedulingInfo.numScheduledTxsSkipped++
+				continue
+			}
+		}
+
+		senderAddressToSkip = tx.GetSndAddr()
+
 		_, txTypeDstShard := txs.txTypeHandler.ComputeTransactionType(tx)
 		isReceiverSmartContractAddress := txTypeDstShard == process.SCDeployment || txTypeDstShard == process.SCInvoking
 		if !isReceiverSmartContractAddress {
 			continue
 		}
 
-		_, alreadyAdded := mapSCTxs[string(txHash)]
-		if alreadyAdded {
+		miniBlock, ok := mapMiniBlocks[receiverShardID]
+		if !ok {
+			log.Debug("scheduled mini block is not created", "shard", receiverShardID)
 			continue
 		}
 
-		log.Debug("this sc call could be added in scheduled mini blocks",
-			"nonce", tx.Nonce,
-			"hash", txHash,
-			"sender shard", senderShardID,
-			"receiver shard", receiverShardID)
+		numNewMiniBlocks := 0
```

Comment:
take some part out into a new function. this is still way too big.

Reply:
WIll be done in the last PR